openstack_sdk/src/config.rs

```diff
@@ -358,51 +461,103 @@ fn get_config_file_search_paths<S: AsRef<str> + std::fmt::Display>(filename: S)
         .flat_map(|x| {
             CONFIG_SUFFIXES
                 .iter()
-                .map(|y| x.join(format!("{}{}", filename, y)))
+                .map(|y| x.join(format!("{}{}", filename.as_ref(), y)))
         })
         .collect();
 }
 
-/// Validate whether single config file is a valid file.
+/// Searches for a `clouds-public.{yaml,yml,json}` config file.
 ///
-/// Returns `true`, `false` with logging information or raises parse error
-fn is_config_valid(config_file: &Path) -> Result<bool, ConfigError> {
-    // Since config lib is not returning information about the file from which the
-    // error comes we try to deserialize each individual file to be user friendly
-    match config::Config::builder()
-        .add_source(File::from(config_file))
-        .build()
-    {
-        Ok(x) => match x.try_deserialize::<ConfigFile>() {
-            Ok(..) => return Ok(true),
-            Err(e) => {
-                error!("Error in file {}: {:?}", config_file.display(), e);
-                return Ok(false);
-            }
-        },
-        Err(x) => return Err(ConfigError::parse(x)),
-    }
+/// The following locations will be tried in order:
+///
+/// - `./clouds-public.{yaml,yml,json}` (current working directory)
+/// - `$XDG_CONFIG_HOME/openstack/clouds-public.{yaml,yml,json}`
+/// - `$XDG_HOME/.config/openstack/clouds-public.{yaml,yml,json}`
+/// - `/etc/openstack/clouds-public.{yaml,yml,json}`
+pub fn find_vendor_file() -> Option<PathBuf> {
```

Comment:
I do not think it makes ever sense to make those pub. Do you have interest using them outside?

Reply:
Personally I do not, but since the builder itself does not add those files and the only way to load them is by using one of the `new` functions, I think it is good to have access to them in order to add those files to the builder if needed.